File: 83_Remove_Duplicates_from_Sorted_List.cpp
#include "iostream"

struct ListNode {
     int val;
     ListNode *next;
     ListNode(int x) : val(x), next(NULL) {}
 };

class Solution {
public:
    ListNode* deleteDuplicates(ListNode* head) {
        ListNode* cur = head;
        while(cur){
            while(cur->next && cur->val == cur->next->val)
                cur->next = cur->next->next;  
            cur = cur->next;
        }

        return head;              
    }
};

File: play/first/83_Remove_Duplicates_from_Sorted_List.py
#!/usr/bin/env python3

# info
# -name   : zhangruochi
# -email  : zrc720@gmail.com


class Solution(object):
    def deleteDuplicates(self, head):
        """
        :type head: ListNode
        :rtype: ListNode
        """
        cur = head
        while cur:
            while cur.next and cur.next.val == cur.val:
                cur.next = cur.next.next
            cur = cur.next

        return head
